### PR TITLE
added Zowe CLI release notes

### PR DESCRIPTION
**Pull request by anaxceron**
Describe your pull request here: adding Zowe CLI release notes

List the file(s) included in this PR: v2_14_0.md

After creating the PR, follow the instructions in the comments.

### CODE REVIEW

File: docs/whats-new/release-notes/v2_14_0.md [hunk 1]
```diff
@@ -47,18 +48,36 @@ Zowe Version 2.14.0 contains the bug fixes that are described in the following t
 
 #### Zowe CLI (Core)
 
+- Corrected extra character being displayed at the end of lines when issuing command `zowe files compare` on Windows. [#1992](https://github.com/zowe/zowe-cli/issues/1992)
+- Corrected the online help description for `zowe files compare uss`. [#1754](https://github.com/zowe/zowe-cli/issues/1754)
+- Fixed typo in command help for `zowe zos-workflows create` commands.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_14_0.md:53]: @adam-wolfe Can we add a PR/issue link to this?

File: docs/whats-new/release-notes/v2_14_0.md [hunk 2]
```diff
@@ -47,18 +48,36 @@ Zowe Version 2.14.0 contains the bug fixes that are described in the following t
...
 
-#### CICS Plug-in for Zowe CLI
+- Fixed error message shown for null option definition to include details about which command caused the error. [#2002](https://github.com/zowe/zowe-cli/issues/2002)
 
 #### DB2 Plug-in for Zowe CLI
 
-#### MQ Plug-in for Zowe CLI
-
-#### IMS Plug-in for Zowe CLI
+- Updated follow-redirects transitive dependency to resolve technical debt.
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_14_0.md:61]: @adam-wolfe Can we add a PR/issue link to this?

File: docs/whats-new/release-notes/v2_14_0.md [hunk 3]
```diff
@@ -47,18 +48,36 @@ Zowe Version 2.14.0 contains the bug fixes that are described in the following t
...
 #### Zowe CLI (Core)
 
+- Corrected extra character being displayed at the end of lines when issuing command `zowe files compare` on Windows. [#1992](https://github.com/zowe/zowe-cli/issues/1992)
+- Corrected the online help description for `zowe files compare uss`. [#1754](https://github.com/zowe/zowe-cli/issues/1754)
+- Fixed typo in command help for `zowe zos-workflows create` commands.
+
 #### Zowe CLI Imperative Framework
 
-#### CICS Plug-in for Zowe CLI
+- Fixed error message shown for null option definition to include details about which command caused the error. [#2002](https://github.com/zowe/zowe-cli/issues/2002)
```
Comment (anaxceron) [inline docs/whats-new/release-notes/v2_14_0.md:57]: @adam-wolfe -- would like to add more detail here. for instance, where is the null option definition done?

File: docs/whats-new/release-notes/v2_14_0.md (hunk 1 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_14_0.md:53]: https://github.com/zowe/zowe-cli/pull/2003

File: docs/whats-new/release-notes/v2_14_0.md (hunk 2 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_14_0.md:61]: https://github.com/zowe/zowe-cli-db2-plugin/pull/139

File: docs/whats-new/release-notes/v2_14_0.md (hunk 3 above)
Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_14_0.md:57]: Option definitions for a command are specified in the Zowe CLI source code (or in Zowe CLI plug-ins) in each command's CommandDefinition. The options are specified as an array of OptionDefinitions. 

If a developer happened to add a null or undefined variable to the options array of a CommandDefinition then build and run Zowe CLI, the CLI used to fail with an unhelpful error message that would print out the entire command tree. With this PR, the error message has been revised to display the command handler file path for the command that failed, which makes it easier for a developer to figure out which CommandDefinition is broken.

Comment (anaxceron) [reply docs/whats-new/release-notes/v2_14_0.md]: I updated the note. Please review for accuracy.

Comment (adam-wolfe) [reply docs/whats-new/release-notes/v2_14_0.md]: Looks good

**Review by adam-wolfe — APPROVED** [review]
LGTM

**Pull request merged**